apps/backend/src/modules/groups/group.entity.ts

```diff
@@ -1,49 +1,52 @@
 import { type Entity } from "~/libs/types/types.js";
-import { type PermissionModel } from "~/modules/groups/permission.model.js";
-import { type UserModel } from "~/modules/users/user.model.js";
 
-import { type GroupCreateResponseDto } from "./libs/types/types.js";
+import { type UserModel } from "../users/user.model.js";
+import {
+	type GroupCreateResponseDto,
+	type GroupGetAllItemResponseDto,
+} from "./libs/types/types.js";
 
 class GroupEntity implements Entity {
+	private createdAt: null | string;
 	private id: null | number;
 	private name!: string;
-	private permissions!: Pick<PermissionModel, "id">[];
+	private permissions!: Array<{ id: number; name?: string }>;
```

Comment:
Why did you change it this way instead of `Pick<PermissionModel, "id" | "name">[]`?

Reply:
I needed to somehow state that name can be optional, since we don't receive it in payload and I can't provide those names to `initializeNew` in `group.service`.  I am now using `Array<Partial<Pick<PermissionModel, "name">> & Pick<PermissionModel, "id">>`.  Can't really find more elegant solution for this.